### PR TITLE
feat: truncate: adds truncate hook and component with overflow utils

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:
- Adds `overflow` utilities `hasHorizontalOverflow`, `hasOverflow` and `hasVerticalOverflow`
- Adds `useTruncate` hook
- Adds `Truncate` component that broadcasts truncation boolean state by id
- Updates `Tooltip` stories to implement `Truncate` and demonstrate only enabling `Tooltip` when necessary
- Adds unit tests
- Exports new features and utilities

https://github.com/EightfoldAI/octuple/assets/99700808/84375139-dc32-46cd-8036-ebc3495eef83

## JIRA TASK (Eightfold Employees Only):
ENG-86929

## CHANGE TYPE:

- [ ] Bugfix Pull Request
- [x] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [x] I have added unittests for this change

## TEST PLAN:
Pull the PR branch and run `yarn` and `yarn storybook`. Verify the `Tooltip` stories behave as expected.

### CODE REVIEW

**Review by ypike-eightfold — APPROVED** [review]

File: src/components/Tooltip/tooltip.module.scss [hunk 1]
```diff
@@ -2,6 +2,7 @@ $tooltip-arrow-shadow: 0px 1px 2px rgba(15, 20, 31, 0.12);
 
 .referenceWrapper {
   display: inline-block;
+  width: fit-content;
```
Comment (jhoward-eightfold) [inline src/components/Tooltip/tooltip.module.scss:5]: any layout risk here?

File: src/hooks/useTruncate.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+'use client';
```
Comment (jhoward-eightfold) [inline src/hooks/useTruncate.tsx:1]: what does this do? SSR related?

File: src/hooks/useTruncate.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    <span
+      {...rest}
+      className={mergeClasses([
+        textClassNames,
+        rest?.className,
+        rest?.classNames,
+      ])}
+      ref={textRef}
+      style={{
+        display: '-webkit-box',
```
Comment (jhoward-eightfold) [inline src/hooks/useTruncate.tsx:66]: need browser prefixes? is this going to be functional cross browser?

**Review by jhoward-eightfold — APPROVED** [review]

File: src/hooks/useTruncate.tsx (hunk 2 above)
Comment (dkilgore-eightfold) [reply src/hooks/useTruncate.tsx:1]: Yes, ensures this is only run on client as it uses DOM.

File: src/hooks/useTruncate.tsx (hunk 3 above)
Comment (dkilgore-eightfold) [reply src/hooks/useTruncate.tsx:66]: Should be fine, webkit is pretty much cross-browser these days anyways. Besides, we've been using this in prod for 2 years now :)

File: src/components/Tooltip/tooltip.module.scss (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Tooltip/tooltip.module.scss:5]: minimal.

**Pull request merged**